proxy/envoy/config.go

```diff
@@ -607,6 +610,81 @@ func buildInboundListeners(mesh *proxyconfig.ProxyMeshConfig, sidecar proxy.Node
 	return listeners, clusters
 }
 
+func appendPortToDomains(domains []string, port int) []string {
+	domainsWithPorts := make([]string, len(domains), 2*len(domains))
+	copy(domainsWithPorts, domains)
+
+	for _, domain := range domains {
+		domainsWithPorts = append(domainsWithPorts, domain+":"+strconv.Itoa(port))
+	}
+
+	return domainsWithPorts
+}
+
+func buildEgressFromSidecarVirtualHostOnPort(rule *proxyconfig.EgressRule, mesh *proxyconfig.ProxyMeshConfig,
+	port *model.Port) *VirtualHost {
+	var externalTrafficCluster *Cluster
+
+	protocolSuffix := "-" + strings.ToLower(string(port.Protocol))
+	if rule.UseEgressProxy {
+		externalTrafficCluster = buildOutboundCluster("istio-egress.default-"+protocolSuffix, port, nil)
+		externalTrafficCluster.ServiceName = ""
+		externalTrafficCluster.Type = ClusterTypeStrictDNS
+		externalTrafficCluster.Hosts = []Host{{URL: fmt.Sprintf("tcp://%s", mesh.EgressProxyAddress)}}
+	} else {
+		externalTrafficCluster = buildOriginalDSTCluster("orig-dst-cluster"+protocolSuffix, mesh.ConnectTimeout)
+		if port.Protocol == model.ProtocolHTTPS {
+			externalTrafficCluster.SSLContext = &SSLContextExternal{}
+		}
+
+		if port.Protocol == model.ProtocolGRPC || port.Protocol == model.ProtocolHTTP2 {
+			externalTrafficCluster.Features = ClusterFeatureHTTP2
+		}
+	}
+
+	externalTrafficRoute := buildDefaultRoute(externalTrafficCluster)
+
+	return &VirtualHost{
+		Name:    rule.Domains[0], // use the first domain as the name
+		Domains: appendPortToDomains(rule.Domains, port.Port),
+		Routes:  []*HTTPRoute{externalTrafficRoute},
+	}
+}
+
+// buildExternalTrafficVirtualHosts builds virtual hosts from egress rule
+func buildEgressFromSidecarVirtualHostsOnPort(rule *proxyconfig.EgressRule, mesh *proxyconfig.ProxyMeshConfig,
+	port *model.Port) []*VirtualHost {
+	hosts := make([]*VirtualHost, 0)
+
+	host := buildEgressFromSidecarVirtualHostOnPort(rule, mesh, port)
+	hosts = append(hosts, host)
+
+	return hosts
+}
```

Comment:
This is unnecessary and just makes it hard to distinguish the two functions. Please fold this for loop into the above function